### PR TITLE
Fix regression test script

### PR DESCRIPTION
**Pull request by jay-zhuang**
Summary: Regression test is broken and not running:
1. failed test is not reporting, fix it by add `set -e`
2. internal regression test is not run inside github, removing that
3. fix a few minor issue to pass the test

Test Plan: CI: https://www.internalfb.com/intern/sandcastle/job/13510799359573861

### CODE REVIEW

File: build_tools/regression_build_test.sh [hunk 1]
```diff
@@ -286,12 +271,10 @@ common_in_mem_args="--db=/dev/shm/rocksdb \
     --sync=0 \
     --verify_checksum=1 \
     --delete_obsolete_files_period_micros=314572800 \
-    --max_grandparent_overlap_factor=10 \
     --use_plain_table=1 \
     --open_files=-1 \
     --mmap_read=1 \
     --mmap_write=0 \
-    --memtablerep=prefix_hash \
```
Comment (ltamasi) [inline build_tools/regression_build_test.sh:294]: Question: why are we removing these two options?

File: build_tools/rocksdb-lego-determinator [hunk 2]
```diff
@@ -379,23 +379,6 @@ LITE_BUILD_COMMANDS="[
...
-        'steps': [
-            $CLEANUP_ENV,
-            {
-                'name':'Report RocksDB Lite binary size to scuba',
-                'shell':'cd $WORKING_DIR; tools/report_lite_binary_size.sh',
-                'user':'root',
-            },
-        ],
-]"
-
```
Comment (ltamasi) [inline build_tools/rocksdb-lego-determinator:398]: I'm assuming we don't have/need such a job anymore, right?

**Review by ltamasi — APPROVED** [review]
LGTM, thanks! Just a couple of questions

File: build_tools/regression_build_test.sh (hunk 1 above)
Comment (jay-zhuang) [reply build_tools/regression_build_test.sh:294]: option `max_grandparent_overlap_factor` is removed. similar PR: https://github.com/facebook/rocksdb/pull/2015

`prefix_hash` doesn't support concurrent writes any more, or we need to set `Options.allow_concurrent_memtable_write` to false.

File: build_tools/rocksdb-lego-determinator (hunk 2 above)
Comment (jay-zhuang) [reply build_tools/rocksdb-lego-determinator:398]: Yeah, the test is never been used. Also our regression test is reporting binary size now, including LITE, shared, static: https://fburl.com/unidash/b9ana9qd

**Pull request closed without merging**